src/core/server/http/http_service.ts

```diff
@@ -117,6 +116,24 @@ export class HttpService
       ) => this.requestHandlerContext!.registerContext(pluginOpaqueId, contextName, provider),
     };
 
+    if (notReadySetup) {
+      this.log.warn('haz not ready');
+      // We cannot use the real context container since the core services may not yet be ready
+      const fakeContext: RequestHandlerContextContainer = deps.context.createContextContainer();
```

Comment:
Optional (if it's simple to implement): it would be great to throw an exception explicitly if someone tries to read `fakeContext` property.

Reply:
Resolved in [`2e0ac39` (#79012)](https://github.com/elastic/kibana/pull/79012/commits/2e0ac399367222177ceed2b4626c21838b63e70f)